ejector/data_api_signing_rate_lookup.go

```diff
@@ -0,0 +1,316 @@
+package ejector
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/validator"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/disperser/dataapi"
+	dataapiv2 "github.com/Layr-Labs/eigenda/disperser/dataapi/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+var _ SigningRateLookup = (*dataApiSigningRateLookup)(nil)
+
+// Uses batch information in dynamoDB to determine signing rates.
+type dataApiSigningRateLookup struct {
+	logger     logging.Logger
+	url        string
+	httpClient *http.Client
+}
+
+// Looks up signing rates from the DataAPI at the given URL.
+func NewDataApiSigningRateLookup(
+	logger logging.Logger,
+	url string,
+	httpTimeout time.Duration,
+) *dataApiSigningRateLookup {
+
+	httpClient := &http.Client{
+		Timeout: httpTimeout,
+	}
+
+	return &dataApiSigningRateLookup{
+		logger:     logger,
+		url:        url,
+		httpClient: httpClient,
+	}
+}
+
+func (srl *dataApiSigningRateLookup) GetSigningRates(
+	timeSpan time.Duration,
+	quorums []core.QuorumID,
+	version ProtocolVersion,
+	omitPerfectSigners bool,
+) ([]*validator.ValidatorSigningRate, error) {
+	switch version {
+	case ProtocolVersionV1:
+		if !omitPerfectSigners {
+			srl.logger.Warn(
+				"omitPerfectSigners flag is ignored for ProtocolVersionV1, will never return perfect signers")
+		}
+		return srl.getV1SigningRates(timeSpan, quorums)
+	case ProtocolVersionV2:
+		return srl.getV2SigningRates(timeSpan, quorums, omitPerfectSigners)
+	default:
+		return nil, fmt.Errorf("unsupported protocol version: %d", version)
+	}
+}
+
+// Look up signing rates for v1.
+func (srl *dataApiSigningRateLookup) getV1SigningRates(
+	timeSpan time.Duration,
+	quorums []core.QuorumID,
+) ([]*validator.ValidatorSigningRate, error) {
+
+	quorumSet := make(map[core.QuorumID]struct{})
+	for _, q := range quorums {
+		quorumSet[q] = struct{}{}
+	}
+
+	now := time.Now()
+
+	path := "api/v1/metrics/operator-nonsigning-percentage"
+	urlStr, err := url.JoinPath(srl.url, path)
+	if err != nil {
+		return nil, fmt.Errorf("error joining URL path with %s and %s: %w", srl.url, path, err)
+	}
+	url, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing URL: %w", err)
+	}
+	// add query parameters
+	q := url.Query()
+	// end: datetime formatted in "2006-01-02T15:04:05Z"
+	q.Set("end", now.UTC().Format("2006-01-02T15:04:05Z"))
```

Comment:
I think you can do this instead of the literal:

`q.Set("end", now.UTC().Format(time.RFC3339))`

(also applies to the v2 method)